domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -3000,13 +3074,22 @@ public void getConnection(Handler<AsyncResult<PgConnection>> replyHandler) {
     });
   }
 
-  void getSQLConnection(Handler<AsyncResult<SQLConnection>> handler) {
+   void getSQLConnection(Handler<AsyncResult<SQLConnection>> handler) {
+    getSQLConnection(handler, 0);
+   }
+
+  void getSQLConnection(Handler<AsyncResult<SQLConnection>> handler, int executionTimeLimit) {
     getConnection(res -> {
       if (res.failed()) {
         handler.handle(Future.failedFuture(res.cause()));
         return;
       }
-      handler.handle(Future.succeededFuture(new SQLConnection(res.result(), null)));
+      SQLConnection sqlConnection = new SQLConnection(res.result(), null, executionTimeLimit);
+      if (executionTimeLimit > 0) {
+         activeConnections.add(sqlConnection);
+      }
+      handler.handle(Future.succeededFuture(sqlConnection));
+      activeConnections.remove(sqlConnection);
```

Comment:
Can you write a unit test that shows that the remove is executed only after the query has completed?

Reply:
Fixed

Reply:
Thank you, the unit test selectWithTimeoutFailure runs the sleep.
But there is no cancel because
* `handler.handle(Future.succeededFuture(sqlConnection))` returns immediately after the query has stated,
* `activeConnections.remove(sqlConnection)` is executed immediately after the query has started.